autogen.sh

```diff
@@ -1,7 +1,33 @@
 #!/bin/bash -e
-[ -d m4 ] || mkdir m4
-libtoolize --copy --force
-aclocal -I m4
-autoconf
-automake --foreign --add-missing --copy
-rm -rf autom4te.cache
+test -n "$srcdir" || srcdir=`dirname "$0"`
+test -n "$srcdir" || srcdir=.
+
+olddir=`pwd`
+cd $srcdir
+
+(test -f configure.ac) || {
+        echo "*** ERROR: Directory "\`$srcdir\'" does not look like the top-level project directory ***"
+        exit 1
+}
+
+PKG_NAME=`autoconf --trace 'AC_INIT:$1' configure.ac`
+if [ "$#" = 0 -a "x$NOCONFIGURE" = "x" ]; then
+        echo "*** WARNING: I am going to run \`configure' with no arguments." >&2
+        echo "*** If you wish to pass any to it, please specify them on the" >&2
+        echo "*** \`$0\' command line." >&2
+        echo "" >&2
+fi
+
+# libtoolize --copy --force --install
+aclocal --install || exit 1
+autoreconf --verbose --force --install -Wno-portability || exit 1
+cd $olddir
```

Comment:
Can't this use ``pushd`` and ``popd`` instead of ``cd`` and remembering the ``$olddir``?